### ISSUE TITLE
Making pages amp compatible

### ISSUE THREAD

**Issue by xRahul** [nielsenramon/chalk#130]
Having amp ready pages will again be a boon for SEO. As the theme is minimal with non-existent dynamic content, we can make amp version of the pages for higher preference in mobile searches.
I have checked out some packages also like [amp-jekyll](https://github.com/juusaw/amp-jekyll). If we can incorporate them, we should do it.
I really like this theme and want to take it to the next level now! I guess going PWA route is too soon though. ;)

**Comment by migueldemoura** [comment]
TL;DR: AMP is a trojan horse, and one ought to be careful before jumping into it with both feet.
Sorry for the long comment :)

AMP's goal was to speed up the horrible multi-MB mobile pages of high traffic publisher/news sites by restricting what they could do and enforcing good practices that weren't being followed. Common, average folks are installing ad and content blockers to fix those painfully slow pages (one of most important factors driving this behavior), which isn't all that great for Google. AMP is also an attempt at locking-in publishers, much like Facebook does with its gamed news feed display of posts that squeeze publishers for more money on every iteration for the same user exposure.

You're correct that it does boost SEO, but that's merely due to Google's dominance, and not as a result of AMP being superior version of the Web™, especially in a properly designed theme like Chalk that doesn't spit 5MB of useless javascript and/or crippling ad code.

This is my opinion: if the objective is purely maximizing SEO - go for it -, but if you want to take the factors I mentioned into account, I'd probably put this functionality behind a config flag.

Having said that, what sort of PWA features would be nice to have on a simple blog theme? The improved app-like behavior is interesting, especially on a pinned icon on mobile. I find myself wanting to do that on some blogs I can't add to my feed reader.

**Comment by xRahul** [comment]
YEah, it was purely for SEO. I know making a web page amp-lified is an undertaking by itself, but having that lightning symbol in the search results would be quite good. I agree that it should be under config and turned on only if required. As there are many things that do not work in amp pages, user should have the option to choose.
Also, I was joking about PWA. I don't see why someone would need to make their blog an app. It would be an optional feature at most. Lets keep that shelved for now though.

**Comment by xRahul** [comment]
I have added both PWA and AMP on my fork of Chalk. It looks great so far.
There are many changes that need to be done, but having AMP version of Posts would be good.
